File: web/src/main/java/by/sorface/passport/web/services/users/DefaultRoleService.java
package by.sorface.passport.web.services.users;

import by.sorface.passport.web.dao.models.RoleEntity;
import by.sorface.passport.web.dao.sql.RoleRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class DefaultRoleService implements RoleService {

    private final RoleRepository roleRepository;

    @Override
    public RoleEntity findByValue(String name) {
        return roleRepository.findByValueIgnoreCase(name).orElse(null);
    }

}

File: web/src/main/kotlin/by/sorface/passport/web/services/users/DefaultRoleService.kt
package by.sorface.passport.web.services.users

import by.sorface.passport.web.dao.sql.models.RoleEntity
import by.sorface.passport.web.dao.sql.repository.RoleRepository
import org.springframework.stereotype.Service

@Service
open class DefaultRoleService(private val roleRepository: RoleRepository) : RoleService {

    override fun findByValue(value: String): RoleEntity? = roleRepository.findByValueIgnoreCase(value).orElse(null)

}
